README.txt
yaaaa mOM

changelog.txt
CREATE COMPUTER SERIAL: BMW45` TAG: 12345 DATE: 2016-04-20 TIME: 10:37:27.411000
CREATE COMPUTER SERIAL: BMW452 TAG: 12345 DATE: 2016-04-20 TIME: 10:38:03.684000
EDIT COMPUTER SERIAL: BMW45` TAG: 11234 SHIP: Feb-4-2010 LOCATION: 2 DATE: 2016-04-20 TIME: 10:38:47.419000
EDIT COMPUTER SERIAL: BMW45` TAG: 12345 SHIP: Feb-4-2010 LOCATION: 2 DATE: 2016-04-20 TIME: 10:39:13.121000
CREATE COMPUTER SERIAL: BMW451 TAG: 12345 DATE: 2016-04-20 TIME: 10:40:19.766000
CREATE COMPUTER SERIAL: BMW112 TAG: 11234 DATE: 2016-04-20 TIME: 10:41:41.884000
CREATE COMPUTER SERIAL: SQL451 TAG: 11234 DATE: 2016-04-20 TIME: 10:44:47.839000
CREATE COMPUTER SERIAL: BMW451 TAG: 11234 DATE: 2016-04-20 TIME: 11:12:04.275000
EDIT COMPUTER SERIAL: BMW112 TAG: 55555 SHIP: Jul-- LOCATION: - DATE: 2016-04-20 TIME: 11:12:41.529000
EDIT COMPUTER SERIAL: BMW451 TAG: 55112 SHIP:  - - - - -  LOCATION:  -  DATE: 2016-04-20 TIME: 11:14:24.166000
EDIT COMPUTER SERIAL: SQL451 TAG: 420420 SHIP:  - - - - -  LOCATION:  -  DATE: 2016-04-20 TIME: 11:20:08.784000
EDIT COMPUTER SERIAL: SQL451 TAG: 420420 SHIP:  -- LOCATION: 1 DATE: 2016-04-20 TIME: 11:21:27.770000
EDIT COMPUTER SERIAL: SQL451 TAG: 11111 SHIP:  -  -  -  LOCATION:  -  DATE: 2016-04-20 TIME: 11:22:54.319000
EDIT COMPUTER SERIAL: SQL451 TAG: 11111 SHIP:  -  -  -  LOCATION:  -  DATE: 2016-04-20 TIME: 11:28:38.079000
CREATE COMPUTER SERIAL: FTW222 TAG: 11111 DATE: 2016-05-04 TIME: 11:23:15.138000
CREATE COMPUTER SERIAL: 123123 TAG: 4125156 DATE: 2016-05-11 TIME: 10:49:21.784000
CREATE COMPUTER SERIAL: 6tgegwet TAG: qweqwr DATE: 2016-05-11 TIME: 10:49:28.811000
CREATE COMPUTER SERIAL: qwrewafasf TAG: asfasdfasd DATE: 2016-05-11 TIME: 10:49:32.681000
CREATE COMPUTER SERIAL: qqq TAG: qqq213 DATE: 2016-05-11 TIME: 10:49:36.926000
CREATE COMPUTER SERIAL: 21415t6wef TAG: qwrqw DATE: 2016-05-11 TIME: 10:49:41.695000
CREATE COMPUTER SERIAL: qwroqwr TAG: qmrqro DATE: 2016-05-11 TIME: 10:49:53.543000
CREATE COMPUTER SERIAL: vbz TAG: zzzz DATE: 2016-05-11 TIME: 10:49:58.612000
CREATE COMPUTER SERIAL: 1235 TAG: ggggg DATE: 2016-05-11 TIME: 10:50:02.767000
CREATE COMPUTER SERIAL: qeqe124 TAG: e2e14 DATE: 2016-05-11 TIME: 10:50:08.851000
CREATE COMPUTER SERIAL: aaaa1 TAG: 412512 DATE: 2016-05-11 TIME: 10:50:13.060000
CREATE COMPUTER SERIAL: aaaaf TAG: fqtqey2351 DATE: 2016-05-11 TIME: 10:50:17.887000
CREATE COMPUTER SERIAL: asfagz TAG: 125ytr DATE: 2016-05-11 TIME: 10:50:22.416000
CREATE COMPUTER SERIAL: zzzz TAG: t67876 DATE: 2016-05-11 TIME: 10:50:26.803000
CREATE COMPUTER SERIAL: nmgi TAG: i159136760 DATE: 2016-05-11 TIME: 10:50:49.343000
CREATE COMPUTER SERIAL: 09878921413 TAG: 1231251 DATE: 2016-05-11 TIME: 10:50:55.975000
CREATE COMPUTER SERIAL: zzz TAG: 125676543 DATE: 2016-05-11 TIME: 10:51:00.211000
CREATE COMPUTER SERIAL: zzzzzzzz TAG: 2156787654 DATE: 2016-05-11 TIME: 10:51:05.007000
CREATE COMPUTER SERIAL: flgjeot TAG: 124o15j DATE: 2016-05-11 TIME: 10:51:12.120000
CREATE COMPUTER SERIAL: ki9123951u5q9rh TAG: rqtith35n52 DATE: 2016-05-11 TIME: 10:51:20.093000
CREATE COMPUTER SERIAL: affarqwe124 TAG: qweqe DATE: 2016-05-11 TIME: 10:51:25.274000
CREATE COMPUTER SERIAL: a TAG: 1 DATE: 2016-05-11 TIME: 10:54:03.979000
CREATE COMPUTER SERIAL: 2 TAG: 3 DATE: 2016-05-11 TIME: 10:54:08.661000
CREATE COMPUTER SERIAL: 2 TAG: 1 DATE: 2016-05-11 TIME: 10:54:13.480000
CREATE COMPUTER SERIAL: 11 TAG: 222 DATE: 2016-05-11 TIME: 10:54:17.449000
CREATE COMPUTER SERIAL: 11456 TAG: az DATE: 2016-05-11 TIME: 10:54:21.905000
CREATE COMPUTER SERIAL: aa TAG: aa DATE: 2016-05-11 TIME: 10:54:25.520000
CREATE COMPUTER SERIAL: aa12 TAG: aa DATE: 2016-05-11 TIME: 10:54:29.562000
CREATE COMPUTER SERIAL: 12456 TAG: zzz DATE: 2016-05-11 TIME: 10:54:34.980000

test.py
import sqlite3
from Tkinter import *

class Application(Frame):
    def say_hi(self):
        print "hi there, everyone!"

    def createWidgets(self):
        self.QUIT = Button(self)
        self.QUIT["text"] = "QUIT"
        self.QUIT["fg"]   = "red"
        self.QUIT["command"] =  self.quit

        self.QUIT.pack({"side": "left"})

        self.hi_there = Button(self)
        self.hi_there["text"] = "Hello",
        self.hi_there["command"] = self.say_hi

        self.hi_there.pack({"side": "left"})

        self.create_entry = Button(self)
        self.create_entry["text"]= "Create Entry",
        self.create_entry["command"]=self.say_hi

        self.create_entry.pack({"side": "left"})
        

    def __init__(self, master=None):
        Frame.__init__(self, master)
        self.pack()
        self.createWidgets()

root = Tk()
app = Application(master=root)
app.mainloop()
root.destroy()


create_test.py
import sqlite3 as lite
import sys

con = lite.connect('test.db')

with con:
    
    cur = con.cursor()    
    cur.execute("CREATE TABLE Computers(Serial TEXT PRIMARY KEY NOT NULL,Tag TEXT  NOT NULL, Ship INT, Location INT);")



gridtest.py
import Tkinter

if __name__ == '__main__':
    form = Tkinter.Tk()

    getFld = Tkinter.IntVar()

    form.wm_title('File Parser')

    stepOne = Tkinter.LabelFrame(form, text=" 1. Enter File Details: ")
    stepOne.grid(row=0, columnspan=7, sticky='W', \
                 padx=5, pady=5, ipadx=5, ipady=5)

    helpLf = Tkinter.LabelFrame(form, text=" Quick Help ")
    helpLf.grid(row=0, column=9, columnspan=2, rowspan=8, \
                sticky='NS', padx=5, pady=5)
    helpLbl = Tkinter.Label(helpLf, text="Help will come - ask for it.")
    helpLbl.grid(row=0)

    stepTwo = Tkinter.LabelFrame(form, text=" 2. Enter Table Details: ")
    stepTwo.grid(row=2, columnspan=7, sticky='W', \
                 padx=5, pady=5, ipadx=5, ipady=5)

    stepThree = Tkinter.LabelFrame(form, text=" 3. Configure: ")
    stepThree.grid(row=3, columnspan=7, sticky='W', \
                   padx=5, pady=5, ipadx=5, ipady=5)

    inFileLbl = Tkinter.Label(stepOne, text="Select the File:")
    inFileLbl.grid(row=0, column=0, sticky='E', padx=5, pady=2)

    inFileTxt = Tkinter.Entry(stepOne)
    inFileTxt.grid(row=0, column=1, columnspan=7, sticky="WE", pady=3)

    inFileBtn = Tkinter.Button(stepOne, text="Browse ...")
    inFileBtn.grid(row=0, column=8, sticky='W', padx=5, pady=2)

    outFileLbl = Tkinter.Label(stepOne, text="Save File to:")
    outFileLbl.grid(row=1, column=0, sticky='E', padx=5, pady=2)

    outFileTxt = Tkinter.Entry(stepOne)
    outFileTxt.grid(row=1, column=1, columnspan=7, sticky="WE", pady=2)

    outFileBtn = Tkinter.Button(stepOne, text="Browse ...")
    outFileBtn.grid(row=1, column=8, sticky='W', padx=5, pady=2)

    inEncLbl = Tkinter.Label(stepOne, text="Input File Encoding:")
    inEncLbl.grid(row=2, column=0, sticky='E', padx=5, pady=2)

    inEncTxt = Tkinter.Entry(stepOne)
    inEncTxt.grid(row=2, column=1, sticky='E', pady=2)

    outEncLbl = Tkinter.Label(stepOne, text="Output File Encoding:")
    outEncLbl.grid(row=2, column=5, padx=5, pady=2)

    outEncTxt = Tkinter.Entry(stepOne)
    outEncTxt.grid(row=2, column=7, pady=2)

    outTblLbl = Tkinter.Label(stepTwo, \
          text="Enter the name of the table to be used in the statements:")
    outTblLbl.grid(row=3, column=0, sticky='W', padx=5, pady=2)

    outTblTxt = Tkinter.Entry(stepTwo)
    outTblTxt.grid(row=3, column=1, columnspan=3, pady=2, sticky='WE')

    fldLbl = Tkinter.Label(stepTwo, \
                           text="Enter the field (column) names of the table:")
    fldLbl.grid(row=4, column=0, padx=5, pady=2, sticky='W')

    getFldChk = Tkinter.Checkbutton(stepTwo, \
                           text="Get fields automatically from input file",\
                           onvalue=1, offvalue=0)
    getFldChk.grid(row=4, column=1, columnspan=3, pady=2, sticky='WE')

    fldRowTxt = Tkinter.Entry(stepTwo)
    fldRowTxt.grid(row=5, columnspan=5, padx=5, pady=2, sticky='WE')

    transChk = Tkinter.Checkbutton(stepThree, \
               text="Enable Transaction", onvalue=1, offvalue=0)
    transChk.grid(row=6, sticky='W', padx=5, pady=2)

    transRwLbl = Tkinter.Label(stepThree, \
                 text=" => Specify number of rows per transaction:")
    transRwLbl.grid(row=6, column=2, columnspan=2, \
                    sticky='W', padx=5, pady=2)

    transRwTxt = Tkinter.Entry(stepThree)
    transRwTxt.grid(row=6, column=4, sticky='WE')

    form.mainloop()


Hunter_College_Library_Systems_Inventory.py
#import modules
import Tkinter as tk
from Tkinter import *
import sqlite3 as lite
from tkMessageBox import*
import datetime,time,sys,ttk,os.path
##list of months/days/years/locations

months_list=[' - ','Jan','Feb','Mar','Apr','May','Jun','Jul',
                            'Aug','Sep','Oct','Nov','Dec']
days_list = [' - ','1','2','3','4','5','6','7','8','9',
                           '10','11','12','13','14','15','16'
                           ,'17','18','19','20','21','22','23','24','25',
                           '26','27','28','29','30','31']
#add years here
years_list =[' - ','2016','2015','2014','2013','2012','2011',
                            '2010','2009','2008','2007','2006','2005','2004',
                            '2003','2002','2001','2000']
#add locations here
locations_list=[' - ','B1','B2','1','2','3','4','5']

#Checks for Computers Database
if os.path.isfile('Computers.db'):
    con = lite.connect('Computers.db',isolation_level=None)
    with con:
        cur = con.cursor()
else:
    #Creates it if its missing; rename db here
    con = lite.connect('Computers.db')

    with con:
            
        cur = con.cursor()
        #edit table parameters here, ex: model
        cur.execute("CREATE TABLE Computers(Serial TEXT NOT NULL,Tag TEXT  NOT NULL, Ship INT, Location INT, PRIMARY KEY(Serial,Tag));")
##Dialog Boxes
   
#Confirmation/Message Boxes Popup
def clickCreate():
    showinfo(title='Entry Created',message="Entry Created")
def clickEdit():
    showinfo(title='Entry Edited',message="Entry Edited")    
#Empty Entry Popup
def emptyEntry():
    showwarning(title="Error",message="Enter Serial/Tag")
#Computer Exists Error Popup
def existEntry():
    showwarning(title="Error",message="Computer Exists Already")
#Computer does not exist Error Popup
def notexistEntry():
    showwarning(title="Error",message="Computer Does Not Exist")
#Display Specific Computer Information from View Entry
def displayInfo():
    showinfo(title='Information',message=output)
    
#Generate Dynamic Summary Page

def summary():
    class Summary(Tk):
        def __init__(self, parent):
            Tk.__init__(self, parent)
            self.title("Summary Page")
            frameOne = tk.LabelFrame(self)
            frameOne.grid(row=0, columnspan=5, rowspan=5,sticky='W', \
                 padx=10, pady=10, ipadx=10, ipady=10)
            cur.execute("SELECT * FROM Computers ORDER BY tag,location")
            rows = cur.fetchall()
            serialgrid=''
            taggrid=''
            shipgrid=''
            locationgrid=''
            for row in rows:
                serialgrid+=row[0]+'\n'
                taggrid+=row[1]+'\n'
                shipgrid+=str(row[2])+'\n'
                locationgrid+=str(row[3])+'\n'
            seriallabel=tk.Label(frameOne,text='Serial',font=LARGE_FONT)
            seriallabel.grid(row=0,column=0)
            serialinformation = tk.Label(frameOne,text=serialgrid)
            serialinformation.grid(row=1,column=0)
            taginformation = tk.Label(frameOne,text=taggrid)
            taginformation.grid(row=1,column=1)
            taglabel=tk.Label(frameOne,text='CUNYTAG',font=LARGE_FONT)
            taglabel.grid(row=0,column=1)
            shipinformation = tk.Label(frameOne,text=shipgrid)
            shipinformation.grid(row=1,column=2)
            shiplabel=tk.Label(frameOne,text='Shipdate',font=LARGE_FONT)
            shiplabel.grid(row=0,column=2)
            locationinformation = tk.Label(frameOne,text=locationgrid)
            locationinformation.grid(row=1,column=3)
            locationlabel=tk.Label(frameOne,text='Location',font=LARGE_FONT)
            locationlabel.grid(row=0,column=3)
    window=Summary(None)
    window.mainloop()

def admin():
    class Admin(Tk):
        def __init__(self,parent):
            Tk.__init__(self,parent)
            self.title("Admin Login")
            
    question=askokcancel(title='Warning',message= 'Only Authorized Users Should Use This')
    if question is True:
        admin=Toplevel()
        admin.title('Admin Login')
        def checklogin():
            user=adminuserentry.get()
            password=adminpassentry.get()
            def delete_entry():
                serial_entry=serial.get()
                serial.delete(0,END)
                cur.execute("SELECT * from Computers where Serial=?",(serial_entry,))
                checkerino= cur.fetchall()
                
                if len(checkerino)!= 0:
                    cur.execute("DELETE FROM Computers WHERE Serial =?",(serial_entry,))
                    showinfo(title="Delete Complete", message="Entry Deleted")
                    writeDelete(serial_entry)
                    
                else:
                    notexistEntry()
                    
            if (user  == 'Poseidon') and (password == 'Trident'):
                admin.destroy()
                adminfunction=Toplevel()
                adminfunction.title=('Admin Functions')
                Label(adminfunction,text="Serial Number").grid(row=0,column=0)
                serial = Entry(adminfunction)
                serial.grid(row=0,column=1)

                delete_button=tk.Button(adminfunction,text="Delete",
                                        command=delete_entry).grid(
                                            row=1,column=0)
                quit_button=tk.Button(adminfunction,text="Quit",
                              command=lambda :adminfunction.destroy()).grid(
                                  row=1,column=1)
            else:
                    showwarning(title='Invalid Username/Password',message=
                                'Invalid Username or Password')
                    
        adminuserlabel=Label(admin,text='Admin User: ').grid(row=0,column=0)
        adminuserentry=Entry(admin)
        adminuserentry.grid(row=0,column=1)
        adminpasslabel=Label(admin,text='Password: ').grid(row=1,column=0)
        adminpassentry=Entry(admin,show="*")
        adminpassentry.grid(row=1,column=1)
        login_button=tk.Button(admin,text="Login",
                               command =lambda :checklogin()).grid(
                                   row=0,column=2)
        quit_button=tk.Button(admin,text="Quit",
                              command=lambda :admin.destroy()).grid(
                                  row=1,column=2)


##Changelog Writing   
   
#write create event to changelog
def writecreate(serial,tag):
    today = str(datetime.date.today())
    currenttime= str(datetime.datetime.now().time())
    changelog=open('changelog.txt','a')
    changelog.write("CREATE COMPUTER SERIAL: " + serial+" TAG: " +tag+" DATE: "+today+
                    " TIME: "+currenttime+"\n")
    changelog.close()
#Write edit event to changelog
def writeEdit(serial,tag,ship,location):
    today = str(datetime.date.today())
    currenttime= str(datetime.datetime.now().time())
    changelog=open('changelog.txt','a')
    changelog.write("EDIT COMPUTER SERIAL: " + serial+" TAG: " +tag+
                    " SHIP: "+str(ship)+" LOCATION: "+str(location)+" DATE: "+today+
                    " TIME: "+currenttime+"\n")
    changelog.close()
    
def writeDelete(serial):
    today = str(datetime.date.today())
    currenttime= str(datetime.datetime.now().time())
    changelog=open('changelog.txt','a')
    changelog.write("DELETE COMPUTER SERIAL: " + serial +" DATE: "+today+
                    " TIME: "+currenttime+"\n")  



LARGE_FONT= ("Times New Roman", 12)
##Linked Frame Container/Object, Preloads all the Entry options upon initiation
class InventoryMGMT(tk.Tk):

    def __init__(self, *args, **kwargs):
        tk.Tk.__init__(self, *args, **kwargs)
        container = tk.Frame(self)
        container.pack(side="top", fill="both", expand = True)
        container.grid_rowconfigure(0, weight=1)
        container.grid_columnconfigure(0, weight=1)
        self.frames = {}
        #All the "Pages" are included here for preloading. Include future pages in this list
        for F in (StartPage, CreatePage, EditPage,EditEntryPage,ViewPage):
            frame = F(container, self)
            self.frames[F] = frame
            frame.grid(row=0, column=0, sticky="nsew")
        self.show_frame(StartPage)
        
        
    def show_frame(self, cont):
        frame = self.frames[cont]
        frame.tkraise()       

##Start Page/Menu Contains all the buttons to access other pages     
class StartPage(tk.Frame):

    def __init__(self, parent, controller):
        tk.Frame.__init__(self,parent)
        frameOne = tk.LabelFrame(self,font=LARGE_FONT)
        frameOne.pack()
        label = tk.Label(frameOne, text="Hunter College Library Inventory Management", font=LARGE_FONT)
        label.pack(pady=10,padx=10)
        create_button = tk.Button(self, text="Create Entry",
                            command=lambda: controller.show_frame(CreatePage))
        create_button.pack()
        edit_button = tk.Button(self, text="Edit Entry",
                            command=lambda: controller.show_frame(EditPage))
        edit_button.pack()
        view_button = tk.Button(self, text ="View Entry",
                            command=lambda: controller.show_frame(ViewPage))
        view_button.pack()
        summary_button =tk.Button(self,text="View Summary",
                                  command=summary)
        summary_button.pack()
        admin_button = tk.Button(self,text = "Admin Login",
                                 command = admin)
        admin_button.pack()
        quit_button=tk.Button(self,text="Quit",
                              command=lambda:controller.destroy())
        quit_button=tk.Button(self,text="Quit",
                              command=lambda:controller.destroy())
        quit_button.pack()

        

##Create Entry Page; Data entry
class CreatePage(tk.Frame):

    def __init__(self, parent, controller):
        tk.Frame.__init__(self, parent)
        frameOne = tk.LabelFrame(self, text=" Create Entry: ",font=LARGE_FONT)
        frameOne.grid(row=0, columnspan=5, rowspan=5,sticky='W', \
                 padx=10, pady=10, ipadx=10, ipady=10)
        Label(frameOne,text="Serial Number").grid(row=1)
        Label(frameOne,text="Tag Number").grid(row=2)
        Label(frameOne,text="Shipdate").grid(row=3)
        Label(frameOne,text="Location").grid(row=4)
        serial = Entry(frameOne)
        tag = Entry(frameOne)
        #Months Entry
        shipmonth =ttk.Combobox(frameOne,width=15)
        shipmonth['values']=months_list
        shipmonth.current(0)
        shipday=ttk.Combobox(frameOne,width=5)
        #Day Entry
        shipday['values']=days_list
        shipday.current(0)
        #Year Entry
        shipyear=ttk.Combobox(frameOne,width=5)
        shipyear['values']=years_list
        shipyear.current(0)
        #Location Entry "Floors"
        location = ttk.Combobox(frameOne,width=5)
        location['values']= locations_list
        location.current(0)
        
        serial.grid(row=1,column=1)
        tag.grid(row=2,column=1)
        shipmonth.grid(row=3,column=1)
        shipday.grid(row=3,column=2)
        shipyear.grid(row=3,column=3)
        location.grid(row=4,column=1)
        #Function to create Computer entry, and store it in database
        
        def create_entry():
            #Gets user input, clears upon button press
            serial_entry=serial.get()
            serial.delete(0,END)
            tag_entry=tag.get()
            tag.delete(0,END)
            ship_entry=shipmonth.get()+'-'+shipday.get()+'-'+shipyear.get()
            shipmonth.current(0)
            shipday.current(0)
            shipyear.current(0)
            location_entry=location.get()
            location.current(0)
           
            
            #Error Catch if Serial or Tag is blank, or if computer exists already no change
            if (serial_entry != '')and (tag_entry != ''):
                try:
                    cur.execute("Insert INTO Computers(Serial,Tag,ship,location) Values(?,?,?,?);",
                                (serial_entry,tag_entry,ship_entry,location_entry))
                    clickCreate()
                    writecreate(serial_entry,tag_entry)
                    lambda:controller.show_frame(CreatePage)
                except:
                    existEntry()                    
            else:
                emptyEntry()
                
        #Menu Buttons
        frameTwo = tk.LabelFrame(self, text= " Menu ",font=LARGE_FONT)
        frameTwo.grid(row=0, column=8,sticky='W', \
                 padx=5, pady=5, ipadx=5, ipady=5)
        enter_button= tk.Button(frameOne, text ="Enter",command = create_entry)
        enter_button.grid(row=4,column=5)
        edit_button = tk.Button(frameTwo, text="Edit Entry",
                           command=lambda: controller.show_frame(EditPage))
        edit_button.grid(row=3,column=2)
        view_button= tk.Button(frameTwo, text="View Entry",
                               command=lambda:controller.show_frame(ViewPage))
        view_button.grid(row=4,column=2)
        return_button = tk.Button(frameTwo, text="Return To Menu",
                            command=lambda: controller.show_frame(StartPage))
        return_button.grid(row=5,column=2)

#Edit Entry Page

class EditPage(tk.Frame):

    def __init__(self, parent, controller):
        tk.Frame.__init__(self, parent)
        frameOne = tk.LabelFrame(self, text=" Edit Entry: ",font=LARGE_FONT)
        frameOne.grid(row=0, columnspan=5, rowspan=5,sticky='W', \
                 padx=10, pady=10, ipadx=10, ipady=10)
        Label(frameOne,text="Serial Number").grid(row=2)
        Label(frameOne,text='AND').grid(row=3)
        Label(frameOne,text="Tag Number").grid(row=4)
        serial = Entry(frameOne)
        tag = Entry(frameOne)
        ship =Entry(frameOne)
        location = Entry(frameOne)
        serial.grid(row=2,column=1)
        tag.grid(row=4,column=1)
        
        def check_entry():
            #Gets user input, clears upon button press
            serial_entry=serial.get()
            serial.delete(0,END)
            tag_entry=tag.get()
            tag.delete(0,END)
            
            #Error Catch if Serial or Tag is blank, or if computer doesn't exist. else opens edit dialog
            if (serial_entry != '')and (tag_entry != ''):
                try:
                    cur.execute("SELECT * from Computers where Serial=? and Tag=?",(serial_entry,tag_entry))
                    rows = cur.fetchall()
                    checkerino=rows[0]
                    
                    global masterserial
                    masterserial = checkerino[0]                                        
                    controller.show_frame(EditEntryPage)
                except:
                    notexistEntry()
                    
            else:
                emptyEntry()

        frameTwo = tk.LabelFrame(self, text= " Menu ",font=LARGE_FONT)
        frameTwo.grid(row=0, column=9,sticky='W', \
                 padx=5, pady=5, ipadx=5, ipady=5)
        #Menu Buttons
        enter_button= tk.Button(frameOne, text ="Enter",command = check_entry)
        enter_button.grid(row=4,column=5)
    
        create_button = tk.Button(frameTwo, text="Create Entry",
                            command=lambda: controller.show_frame(CreatePage))
        create_button.grid(row=3,column=2)

        view_button= tk.Button(frameTwo, text="View Entry",
                               command=lambda:controller.show_frame(ViewPage))
        view_button.grid(row=4,column=2)

        return_button = tk.Button(frameTwo, text="Return To Menu",
                            command=lambda: controller.show_frame(StartPage))
        return_button.grid(row=5,column=2)
##Actual Edit Dialog. Data Entered will be changed upon user confirmation.

class EditEntryPage(tk.Frame):
    def __init__(self, parent, controller):
        tk.Frame.__init__(self, parent)
        frameOne = tk.LabelFrame(self, text=" Edit Entry: *Items left Blank will be Unchanged",font=LARGE_FONT)
        frameOne.grid(row=0, columnspan=5, rowspan=5,sticky='W', \
                 padx=10, pady=10, ipadx=10, ipady=10)
        
        Label(frameOne,text="Tag Number").grid(row=3)
        Label(frameOne,text="Shipdate").grid(row=4)
        Label(frameOne,text="Location").grid(row=5)
        tag = Entry(frameOne)
        
        tag.grid(row=3,column=1)
        #Months Entry
        shipmonth =ttk.Combobox(frameOne,width=15)
        shipmonth['values']=months_list
        shipmonth.current(0)
        shipday=ttk.Combobox(frameOne,width=5)
        #Day Entry
        shipday['values']=days_list
        shipday.current(0)
        #Year Entry
        shipyear=ttk.Combobox(frameOne,width=5)
        shipyear['values']=years_list
        shipyear.current(0)
        #Location Entry "Floors"
        location = ttk.Combobox(frameOne,width=5)
        location['values']=locations_list
        location.current(0)
        shipmonth.grid(row=4,column=1)
        shipday.grid(row=4,column=2)
        shipyear.grid(row=4,column=3)
        location.grid(row=5,column=1)
        

        def check_entry():
            #Gets user input, clears upon button press, Executes EDIT ENTRY
            tag_entry=tag.get()
            tag.delete(0,END)
            ship_entry=shipmonth.get()+shipday.get()+shipyear.get()
            shipmonth.delete(0,END)
            shipday.delete(0,END)
            shipyear.delete(0,END)
            location_entry=location.get()
            location.delete(0,END)
            cur.execute("SELECT * FROM Computers where Serial = ?",[masterserial])
            rows= cur.fetchall()
            checkerino=rows[0]
            #If Patron leaves entry blank = NO CHANGE
            if tag_entry=='':
                tag_entry=checkerino[1]
            if ship_entry==' - '+ ' - ' + ' - ':    
                ship_entry=checkerino[2]
            if location_entry=='-':
                location_entry=checkerino[3]
            cur.execute("UPDATE Computers SET tag =?,ship=?,location=? WHERE serial=?",
                        (tag_entry,ship_entry,location_entry,masterserial))
            clickEdit()
            writeEdit(masterserial,tag_entry,ship_entry,location_entry)
            controller.show_frame(EditPage)


        enter_button= tk.Button(frameOne, text ="Enter",command = check_entry)
        enter_button.grid(row=5,column=2)



        
class ViewPage(tk.Frame):
    def __init__(self, parent, controller):
        tk.Frame.__init__(self, parent)
        frameOne = tk.LabelFrame(self, text=" View Entry: ",font=LARGE_FONT)
        frameOne.grid(row=0, columnspan=5, rowspan=5,sticky='W', \
                 padx=10, pady=10, ipadx=10, ipady=10)
        Label(frameOne,text="Serial Number").grid(row=2)
        Label(frameOne,text='OR').grid(row=3)
        Label(frameOne,text="Tag Number").grid(row=4)
        
        serial = Entry(frameOne)
        tag = Entry(frameOne)
        serial.grid(row=2,column=1)
        tag.grid(row=4,column=1)
        
        def view_entry():
            #Gets user input, clears upon button press
            serial_entry=serial.get()
            serial.delete(0,END)
            tag_entry=tag.get()
            tag.delete(0,END)
            
            #Error Catch if both Serial or Tag are blank
            if (serial_entry != '')or (tag_entry != ''):
                try:
                    cur.execute("SELECT * from Computers where Serial=? or Tag=?",(serial_entry,tag_entry))
                    rows = cur.fetchall()
                    if len(rows)== 0:
                        notexistEntry()
                        
                    else:
                        for row in rows:
                            #Displays info for a single/unique computer in a popup messagebox   
                            if len(rows)==1:
                                global output
                                convert=rows[0]
                                ttserial= "Serial: "+convert[0]
                                tttag = "\nTag: " + convert[1]
                                ttship="\nShip: " + str(convert[2])
                                ttlocation="\nLocation: " +str(convert[3])
                                
                                output= ttserial+tttag+ttship+ttlocation
                                           
                                displayInfo()
                            else:
                                #Displays multiple boxes for computers with the same tag.... 
                                ttserial= "Serial: "+row[0]
                                tttag = "\nTag: " + row[1]
                                ttship="\nShip: " + str(row[2])
                                ttlocation="\nLocation: " +str(row[3])
                                output= ttserial+tttag+ttship+ttlocation
                                displayInfo()
                except:
                    notexistEntry()   
            else:
                emptyEntry()
        #Menu Buttons
        frameTwo = tk.LabelFrame(self, text= " Menu ",font=LARGE_FONT)
        frameTwo.grid(row=0, column=8,sticky='W', \
                 padx=5, pady=5, ipadx=5, ipady=5)
        enter_button= tk.Button(frameOne, text ="Enter",command = view_entry)
        enter_button.grid(row=4,column=2)
        create_button = tk.Button(frameTwo, text="Create Entry",
                            command=lambda: controller.show_frame(CreatePage))
        create_button.grid(row=3,column=2)
        edit_button = tk.Button(frameTwo, text="Edit Entry",
                            command=lambda: controller.show_frame(EditPage))
        edit_button.grid(row=4,column=2)
        return_button = tk.Button(frameTwo, text="Return To Menu",
                            command=lambda: controller.show_frame(StartPage))
        return_button.grid(row=5,column=2)
    


app = InventoryMGMT()
app.wm_title("Hunter College Library Systems Inventory Management")
app.mainloop()
